tensorflow/compiler/mlir/hlo/lib/Analysis/userange_analysis.cc

```diff
@@ -255,25 +253,21 @@ void UseInterval::intervalSubtract(UseInterval::Vector &a,
     // iterA is strictly before iterB => increment iterA.
     if (*iterA < *iterB) {
       ++iterA;
-    } else if (iterA->start == iterB->start && iterA->end > iterB->end) {
-      // Usually, we would expect the case of iterB beeing strictly before
-      // iterA. However, due to the initial assumption that all intervals of b
-      // are included in some interval of a, we do not need to check if iterB is
-      // strictly before iterA.
-      // iterB is at the start of iterA, but iterA has some values that go
-      // beyond those of iterB. We have to set the lower bound of iterA to the
-      // upper bound of iterB + 1 and increment iterB.
-      // A(3, 100) - B(3, 5) => A(6,100)
+      // iterB is strictly before iterA => increment iterB.
```

Comment:
I find it strange where the comments are. Either they are all in the branch to describe which case it handles or all before. I think the latter is more readable.